### PR TITLE
fix(coderd/debug): fix caching issue with dismissed sections

### PR DESCRIPTION
**Pull request by johnstcn**
Previously we were making healthcheck sections as dismissed in the healthcheck function, the reuslt of which gets cached. This commit moves the logic outside of the function. It results in an extra database lookup on every call to healthcheck.

### CODE REVIEW

File: coderd/debug.go [hunk 1]
```diff
@@ -269,7 +298,7 @@ func loadDismissedHealthchecks(ctx context.Context, db database.Store, logger sl
 		}
 	}
 	if err != nil && !xerrors.Is(err, sql.ErrNoRows) {
-		logger.Error(ctx, "unable to fetch health settings: %w", err)
+		logger.Error(ctx, "unable to fetch health settings", slog.Error(err))
```
Comment (johnstcn) [inline coderd/debug.go:301]: review: drive-by: this causes slog to panic

Comment (mtojek) [reply coderd/debug.go:301]: out of curiosity: why?

File: coderd/debug.go [hunk 2]
```diff
@@ -89,12 +94,36 @@ func (api *API) debugDeploymentHealth(rw http.ResponseWriter, r *http.Request) {
...
+		case codersdk.HealthSectionWebsocket:
+			hc.Websocket.Dismissed = true
+		case codersdk.HealthSectionWorkspaceProxy:
+			hc.WorkspaceProxy.Dismissed = true
+		}
+	}
+
 	format := r.URL.Query().Get("format")
 	switch format {
 	case "text":
```
Comment (mtojek) [inline coderd/debug.go:129]: not part of this PR: we could update it to use severity + dismissed as well.

Comment (mtojek) [inline coderd/debug.go:129]: not part of this PR: "simple" is not an available option?

File: coderd/debug.go [hunk 3]
```diff
@@ -89,12 +94,36 @@ func (api *API) debugDeploymentHealth(rw http.ResponseWriter, r *http.Request) {
...
+			})
+			return
+		}
+		formatHealthcheck(ctx, rw, r, *report, dismissed...)
 		return
 	}
 }
 
-func formatHealthcheck(ctx context.Context, rw http.ResponseWriter, r *http.Request, hc *healthcheck.Report) {
+func formatHealthcheck(ctx context.Context, rw http.ResponseWriter, r *http.Request, hc healthcheck.Report, dismissed ...codersdk.HealthSection) {
```
Comment (mtojek) [inline coderd/debug.go:110]: nit: if you have switch to go over a finite number of sections, this does not have to be varags.

File: coderd/debug.go [hunk 4]
```diff
@@ -89,12 +94,36 @@ func (api *API) debugDeploymentHealth(rw http.ResponseWriter, r *http.Request) {
...
+	// Mark any sections previously marked as dismissed.
+	for _, d := range dismissed {
```
Comment (mtojek) [inline coderd/debug.go:112]: I'm afraid that somebody may forget to extend this switch in the future. Maybe we should move the logic to codersdk, closer to Healthcheck report?

**Review by mtojek — APPROVED** [review]
Nothing blocking

File: coderd/debug.go (hunk 3 above)
Comment (johnstcn) [reply coderd/debug.go:110]: Will address in a follow-up.

File: coderd/debug.go (hunk 4 above)
Comment (johnstcn) [reply coderd/debug.go:112]: Will address in a follow-up.

File: coderd/debug.go (hunk 2 above)
Comment (johnstcn) [reply coderd/debug.go:129]: I thought the available formats were just `json` and `text`.

Comment (johnstcn) [reply coderd/debug.go:129]: Will address in a follow-up.
